index.js

```diff
@@ -115,12 +114,51 @@ var isPercyEnabled = true;
 module.exports = {
   name: 'ember-percy',
 
+  _clientInfo: function() {
+    if(!this._clientInfoCache) {
+      var version = require('./package.json').version;
+      this._clientInfoCache = `${this.name}/${version}`;
+    }
+
+    return this._clientInfoCache;
+  },
+
+  _environmentInfo: function() {
+    if(!this._environmentInfoCache) {
+      this._environmentInfoCache = [
+        `ember/${this._emberSourceVersion()}`,
+        `ember-cli/${this._emberCliVersion()}`
+      ].filter(function(el) {
+        return !el.match(/\/$/);
+      }).join('; ');
+    }
+
+    return this._environmentInfoCache;
+  },
+
+  _emberSourceVersion: function() {
+    try {
+      return require('ember-source/package.json').version;
+    } catch (e) {
+      return '';
```

Comment:
What do you think about making these return `unknown` or `error` so that we get `ember/unknown` or `ember-cli/unknown` (since in this Ember world we should always have a version number for them otherwise something is wrong), and dropping the `filter` above that removes blank things?